src/VisualStudio/IntegrationTest/TestSetup/IntegrationTestServiceCommands.cs

```diff
@@ -121,11 +120,9 @@ private void StartServiceCallback(object sender, EventArgs e)
 
                 // Async initialization is a workaround for deadlock loading ExtensionManagerPackage prior to
                 // https://devdiv.visualstudio.com/DevDiv/_git/VSExtensibility/pullrequest/381506
-                _ = ThreadHelper.JoinableTaskFactory.RunAsync(async () =>
+                _ = Task.Run(async () =>
                 {
-                    await TaskScheduler.Default;
-
-                    var componentModel = await AsyncServiceProvider.GlobalProvider.GetServiceAsync<SComponentModel, IComponentModel>().ConfigureAwait(false);
+                    var componentModel = (IComponentModel?)await AsyncServiceProvider.GlobalProvider.GetServiceAsync(typeof(SComponentModel)).ConfigureAwait(false);
```

Comment:
whis is this safe, if we're running in the bG?  doesn't the cast have the same issue?

Reply:
This is an old integration test outlier. It will eventually be removed by the migration to the new integration test project. In the meantime, it leverages the fact that `IComponentModel` is implemented by a managed service. If this was production code, I would require documentation that the `SComponentModel` service returns a free-threaded object before making this change.

Reply:
This was a good catch. There were a couple other places where I switched to the non-generic form prior to adding `RoslynServiceExtensions`, so I audited the PR for those and will include the fixes in the next push.

Reply:
sounds good.  :)